x/distribution/keeper/allocation.go

```diff
@@ -79,22 +79,45 @@ func (k Keeper) AllocateTokens(
 
 	// calculate fraction allocated to validators
 	communityTax := k.GetCommunityTax(ctx)
-	voteMultiplier := sdk.OneDec().Sub(proposerMultiplier).Sub(communityTax)
+	foundationTax := k.GetSecretFoundationTax(ctx)
 
-	// allocate tokens proportionally to voting power
-	// TODO consider parallelizing later, ref https://github.com/enigmampc/cosmos-sdk/pull/3099#discussion_r246276376
+	// Define the vote multiplier as (100 - proposerMultiplier - communityTax - foundationTax)%.
+	// The value along with a validator's power fraction, is used to determine the
+	// validator's rewards. Note, foundationTax may be zero.
+	voteMultiplier := sdk.OneDec().Sub(proposerMultiplier).Sub(communityTax).Sub(foundationTax)
+
+	var foundationTaxSum sdk.DecCoins
+	if !foundationTax.IsZero() {
+		foundationTaxSum = feesCollected.MulDecTruncate(foundationTax)
+		remaining = remaining.Sub(foundationTaxSum)
+	}
+
+	// allocate tokens proportionally to voting power minus any taxes
 	for _, vote := range previousVotes {
 		validator := k.stakingKeeper.ValidatorByConsAddr(ctx, vote.Validator.Address)
 
-		// TODO consider microslashing for missing votes.
-		// ref https://github.com/enigmampc/cosmos-sdk/issues/2525#issuecomment-430838701
 		powerFraction := sdk.NewDec(vote.Validator.Power).QuoTruncate(sdk.NewDec(totalPreviousPower))
 		reward := feesCollected.MulDecTruncate(voteMultiplier).MulDecTruncate(powerFraction)
+
+		// allocate tokens to the validator
 		k.AllocateTokensToValidator(ctx, validator, reward)
+
+		// update the remaining allocation for the community pool
 		remaining = remaining.Sub(reward)
 	}
 
-	// allocate community funding
+	// Send the foundation tax sum to the foundation tax address. Note, the taxes
+	// collected are decimals and when coverted to integer coins, we must truncate.
+	// The remainder is given back to the community pool.
+	foundationTaxAddr := k.GetSecretFoundationAddr(ctx)
+	foundationTaxSumTrunc, rem := foundationTaxSum.TruncateDecimal()
+	remaining = remaining.Add(rem...)
+
+	if err := k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, foundationTaxAddr, foundationTaxSumTrunc); err != nil {
+		panic(err)
```

Comment:
What happens if `foundationTaxAddr` is not set?
Does a panic here means a chain halt?
Instead of a panic, can we send the foundation tax to the community pool?